### PR TITLE
Revised ImageBol_Model_Downloader edition

### PR DESCRIPTION
**Pull request by ravi5175**
### **Bug Fixes**
-bug fixed where download gets interrupted after a complete file download
-bug fixed where imagebol_model_downloader needs to be called again to start downloading model
-bug fixed where model downloaded to current working directory irrespective of the installation directory
-bug fixed where windows user experience executive deadlock state during writing response

###**Improvements**
-Improved Logic with optimized performance in some cases
-cleaned code and added a status_code information chart with debugger

### CODE REVIEW

File: freshlybuiltimagebol/downloader.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+"""status_code_meanings"""
+# 1000 - model already exist
+# 1001 - model name incorrect
+# 1002 - download interupted
+# 1003 - successful download
+# 1004 - http error
+# 1005 - connection error
+# 1006 - timeout error
+# 1007 - miscellanious error
+# 1008 - building models directory failed
```
Comment (vishal2612200) [inline freshlybuiltimagebol/downloader.py]: Put this docstring in the starting of the class for better readability

File: freshlybuiltimagebol/downloader.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                                f.write(response.content)
+                            else:
+                                chunk_size=1024
+                                for data in tqdm(iterable = response.iter_content(chunk_size),total=total_length/chunk_size, unit = 'KB'):
+                                    try:
+                                        f.write(data)
+                                    except:
+                                        pass
+                            
+
```
Comment (vishal2612200) [inline freshlybuiltimagebol/downloader.py:49]: Remove extra empty lines.. from the file.

File: freshlybuiltimagebol/downloader.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+# 1006 - timeout error
+# 1007 - miscellanious error
+# 1008 - building models directory failed
+      
+
+""" testing area"""
+#model_name=input("model name: ")
+#print(imagebol_model_downloader(model_name).status_code)           
+
+
```
Comment (vishal2612200) [inline freshlybuiltimagebol/downloader.py]: Remove these extra empty lines

**Comment by KapilBansal320** [comment]
Add a gitignore file to ignore cache files

**Pull request merged**